### ISSUE TITLE
Custom Claim Containing Table Value

### ISSUE THREAD

**Issue by mrrobby** [SkyLothar/lua-resty-jwt#49]
I've been working though [auth0/nginx-jwt](https://github.com/auth0/nginx-jwt) and following what you've done since with jwt-validators, in my own attempt to add one more bit of functionality.  I wanted to propose an addition or see if I missed something in my own implementation.

**Motivation**: validating a claim containing an array/table of values from which one must match.  For example, validating `Foo` as a role in a token containing something of the form:

```
{
  "roles" : [
     "Foo",
     "Bar"
   ],
   iss: "issuer",
   .
   .
},
```

I was able to accomplish this combining the `validators.any_of` with the `nginx-jwt` method for `table_contains`, looking something like:

```
access_by_lua_block {
     jwt = require("auth.nginx-jwt")
     local valid_roles = {"Foo","Bar"}
     .
     .
     local claim_spec = {
         .
         roles = validators.any_of(valid_roles, jwt.table_contains, "roles","table","string")
         .
     }
     .
     .
}
```

So, I'm proposing a method that uses `table_contains` in a similar form of `equality_function` and its friends, altogether allowing to expose a new validator:

```
validators.contains_any_of(check_values)(opt)
```

Thoughts?

**Comment by SkyLothar** [comment]
👍 Great idea, I'll make it happen in few days.
PR is welcome.

**Comment by mrrobby** [comment]
Sounds good @SkyLothar .  I have some updates I can push.  The test methods may need some review and running as my dev machine here can't run docker, but the addition was relatively straightforward.

**Comment by SkyLothar** [comment]
You can pull request, and [traivs](https://travis-ci.org/SkyLothar/lua-resty-jwt) will run the test case for you.

**Comment by mrrobby** [comment]
I can do that.  I've been trying, but it's github is not authorizing my push.  Do you have any restrictions on this repo?  I've checked and verified my ssh keys (even regenerated a new one), using a git for remote returns
```
lua-resty-jwt mrrobby$ git push -u origin feature/contains_any_of
ERROR: Permission to SkyLothar/lua-resty-jwt.git denied to mrrobby.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I tried using https instead and that returns a `403` error.  All typical checks have been made.  Thoughts?

**Comment by SkyLothar** [comment]
Oh, Github doesnot work like that.

To make a contribution
1. You have to fork this repo first, creating `mrrobby/lua-resty-jwt`.
2. Submit a PR from `mrrobby/lua-resty-jwt@feature-branch` to `SkyLothar/lua-resty-jwt@master`

**Comment by mrrobby** [comment]
Sorry about that.  See PR #52

**Comment by lucafavatella** [comment]
This should be closed by #52 I believe.
